dlp/dlp_snippets/main.go

```diff
@@ -0,0 +1,304 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// dlp is an example of using the DLP API.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"strings"
+
+	"golang.org/x/net/context"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"github.com/fatih/color"
+	dlppb "google.golang.org/genproto/googleapis/privacy/dlp/v2"
+)
+
+type minLikelihoodFlag struct {
+	l dlppb.Likelihood
+}
+
+func (m *minLikelihoodFlag) String() string {
+	return fmt.Sprint(m.l)
+}
+
+func (m *minLikelihoodFlag) Set(s string) error {
+	l, ok := dlppb.Likelihood_value[s]
+	if !ok {
+		return fmt.Errorf("not a valid likelihood: %q", s)
+	}
+	m.l = dlppb.Likelihood(l)
+	return nil
+}
+
+func minLikelihoodValues() string {
+	var s []string
+	for _, m := range dlppb.Likelihood_name {
+		s = append(s, m)
+	}
+	return strings.Join(s, ", ")
+}
+
+type bytesTypeFlag struct {
+	bt dlppb.ByteContentItem_BytesType
+}
+
+func (f *bytesTypeFlag) String() string {
+	return fmt.Sprint(f.bt)
+}
+
+func (f *bytesTypeFlag) Set(s string) error {
+	b, ok := dlppb.ByteContentItem_BytesType_value[s]
+	if !ok {
+		return fmt.Errorf("not a valid BytesType: %q", s)
+	}
+	f.bt = dlppb.ByteContentItem_BytesType(b)
+	return nil
+}
+
+func bytesTypeValues() string {
+	var s []string
+	for _, m := range dlppb.ByteContentItem_BytesType_name {
+		s = append(s, m)
+	}
+	return strings.Join(s, ", ")
+}
```

Comment:
this is all a little bit too tricky for my liking. if you're at all leaning toward just parsing these in main(), then go for it. if not, then this is fine.

Reply:
Changed to be parsed in main. This is definitely easier to understand.